src/controllers/account/index.ts

```diff
@@ -1,45 +1,68 @@
 import { BaseContext } from 'koa'
-// import AccountUtilities from './account'
+import * as Strings from '../../helpers/strings'
+import * as Methods from '../../helpers/methods'
+import {
+  getStoredVerifierHash,
+  generateAndStoreToken
+} from './account'
 
 export default class AccountController {
-  public static async requestReset (ctx: BaseContext) {
+  // Endpoint to create hash to send in email to the user on login
+  public static async login (ctx: BaseContext) {
     const { email } = ctx.request.body
-    ctx.assert(email, 400)
-    // TODO see if email address is in DB, if not: bail
-    // const resetToken = await AccountUtilities.generateResetToken()
-    // const verifierHash = await AccountUtilities.getVerifierHash(resetToken)
-    // await AccountUtilities.storeResetToken(resetToken.selector, verifierHash)
-    // const emailToken = await AccountUtilities.getEmailResetToken(resetToken)
-    // // TODO email token to user
+    ctx.assert(email, 400, Strings.EmailIsRequired)
 
-    ctx.status = 200
-    ctx.body = { ok: true }
-  }
-  public static async receiveReset (ctx: BaseContext) {
-    const { token } = ctx.request.body
-    ctx.assert(token, 400)
-    // const resetToken = await AccountUtilities.splitSelectorVerifier(token)
-    // DB lookup first 16 bytes provided (selector)
-    // if found, hash next 16 bytes provided (verifier)
-    // const verifierHash = await AccountUtilities.getVerifierHash(resetToken)
-    // compare hash with verifier hash in DB
+    // Generate token, store it
+    const emailToken = await generateAndStoreToken(ctx, false)
     try {
-      // const storedVerifierHash = await AccountUtilities.getStoredVerifierHash(resetToken)
-      // const match = await AccountUtilities.compareHashes(storedVerifierHash, verifierHash)
-      // if (!match) {
-      //   // if hash does not match, 401 and delete token
-      //   await AccountUtilities.deleteSelector(resetToken)
-      //   ctx.assert(match, 401)
-      // }
+      await Methods.sendMagicLinkEmail(email, emailToken)
+      ctx.body = { ok: true }
     } catch (e) {
-      ctx.throw(401)
+      console.log(e)
+      ctx.body = { ok: false, message: e.message || 'unknown error' }
     }
+  }
+
+  // Endpoint to create hash to send in email to the user on register
+  public static async register (ctx: BaseContext) {
+    const { email, role } = ctx.request.body
+    ctx.assert(email, 400, Strings.EmailIsRequired)
+    ctx.assert(role, 400, Strings.RoleIsRequired)
 
-    // if match and hasn't expired, reset the password
-    return AccountController.resetPassword(ctx)
+    // Generate token, store it
+    const emailToken = await generateAndStoreToken(ctx, false)
```

Comment:
This route should differ from the login route by storing more than just the email+token pair+expiration. It should store the data needed to create the user upon verification that the email address is valid. Whatever was sent up in the registration form should be stored in Dynamo, either in a JSON blob or in separate fields. Wouldn't hurt to have a separate table for registration and login items so things don't get muddied.

Reply:
added in a registered field to the record so we know if the account has been registered or not